### PR TITLE
New Object: Enemy Blocker (Updated)

### PR DESCRIPTION
**Pull request by Karkus476**
Blocks badguys, like invisible walls, but not Tux.
Original code by **Kovago Zoltan**. *(who?)*
Submitted by @Savsish originally in #457.
I have rebased his commits for him, and pulled from my fork.

**Disclaimer:** *I haven't tested this*, and this pull request is only to secure the code changes Savs submitted, in case anything happens at his end. Furthermore, he appears to have been banned from this repository, so he cannot contribute further.

@Hume2 You commented on his commit before. Please could you do the same for this one here, and if you wish to fix it, I can push my branch to the SuperTux repo, so that you can do so.

Please see SuperTux/data#15

Edit by @Karkus476: 9th August 2016: Correct mentions and emboldening/italics

### CODE REVIEW

File: src/object/enemy_blocker.cpp [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+EnemyBlocker::after_editor_set() {
+  bbox.set_size(width, height);
+}
+
+HitResponse
+EnemyBlocker::collision(GameObject& other, const CollisionHit& )
+{
+  BadGuy* badguy = dynamic_cast<BadGuy*> (&other);
+  
+  if (badguy == 0)
```
Comment (Hume2) [inline src/object/enemy_blocker.cpp]: This should be done via the function collides. Is it so hard to get it?

**Comment by maxteufel** [comment]
Can this class an the InvisibleWall class be generalized to a common Blocker class from which both classes will inherit then? That could save us some lines of code.

**Comment by maxteufel** [comment]
@SuperTux/level-designers is this feature useful for levels at all?

**Comment by Rusty-Box** [comment]
@maxteufel I would say no and I'm not sure why we would need an enemy blocker at all.

**Comment by maxteufel** [comment]
I suggest we close this then.

**Comment by maxteufel** [comment]
If no one objects to closing this because this seems to be an unnecessary feature, I will close this tomorrow.

**Comment by qwertychouskie** [comment]
I think this might be useful for some levels, currently the only enemy that can go back and forth in a fixed range is the [Crystallo](http://supertux.lethargik.org/wiki/Crystallo).

**Comment by maxteufel** [comment]
10:53:22 <+mteufel[m]> RustyBox: do you think https://github.com/SuperTux/supertux/pull/536#issuecomment-244474582 is a valid reason to add enemy blocker?
10:54:50 <+mteufel[m]> I'm not exactly comfortable with enemy blocker if you say you don't need it. It also just adds more code to maintain. And I don't think that trading that off for add-on designers is a good idea.
10:55:09 <+mteufel[m]> Plus, it might confuse users
10:58:42 <RustyBox[m]> Max Teufel: I agree. The enemy blocker might be confusing, because for example: you would never expect a mr.snowball to turn around without running against a wall.
10:59:16 <+mteufel[m]> Closing then.
10:59:40 <RustyBox[m]> I would say so

**Pull request closed without merging**